app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -399,6 +399,18 @@ class UserSession @Inject constructor(
   fun syncAndClearData(patientRepository: PatientRepository, syncRetryCount: Int = 0, timeoutSeconds: Long = 15L): Completable {
     Timber.i("Syncing and clearing all patient related data")
 
+    val updateUserStatusToResettingPin = requireLoggedInUser()
+        .firstOrError()
+        .flatMapCompletable { user -> updateLoggedInStatusForUser(user.uuid, RESETTING_PIN) }
```

Comment:
This is another hidden behavior inside a method that should only be responsible for clearing data. 
This should have been a separate method to be invoked from the caller itself. In this case `ForgotPinConfirmPinScreenController`. 
No need to refactor it in this PR. Just an FYI.

Every refactor in this class leads to more cracks opening :(

Reply:
Good point. I decided to refactor this and improve the tests a little bit.